src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs

```diff
@@ -0,0 +1,570 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.UI.ViewManagement;
+using Microsoft.Extensions.Logging;
+using Uno.Extensions;
+using Uno.UI;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Media.Animation;
+using XamlWindow = Microsoft.UI.Xaml.Window;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Animation;
+using XamlWindow = Windows.UI.Xaml.Window;
+#endif
+
+namespace Uno.Toolkit.UI
+{
+	/// <summary>
+	/// Provides attached properties on any <see cref="FrameworkElement"/> or can be used as a standalone control which automatically
+	/// adds padding/margin to a control that ensures its content will always be inside 
+	/// the <see cref="ApplicationView.VisibleBounds"/> of the application. SafeArea provides <see cref="InsetMask.SoftInput"/> that will
+	/// apply the proper amount of margin or padding to ensure that the control is never occluded by the on-screen keyboard.
+	/// </summary>
+	public partial class SafeArea : ContentControl
+	{
+		[Flags]
+		public enum InsetMask
+		{
+			All = Left | Right | Top | Bottom,
+			None = 0,
+			Top = 1,
+			Bottom = 2,
+			Left = 4,
+			Right = 8,
+			SoftInput = 16
```

Comment:
It can be, I was thinking I don't want to touch the behavior of All so it would be easier when transitioning from `VisibleBoundsPadding.PaddingMask="All"` to `SafeArea.Insets="All"`

But you think it would be better to have a VisibleBounds and an All that includes SoftInput? And we just document the change in the section of the docs that will highlight migrating from VBP to SafeArea?